worldedit-core/src/main/java/com/sk89q/worldedit/extension/factory/parser/DefaultBlockParser.java

```diff
@@ -370,87 +360,60 @@ private BaseBlock parseLogic(String input, ParserContext context) throws InputPa
-            return validate(context, new SignBlock(state, text));
+            return new SignBlock(state, text);
         } else if (blockType == BlockTypes.SPAWNER) {
             // Allow setting mob spawn type
             if (blockAndExtraData.length > 1) {
                 String mobName = blockAndExtraData[1];
                 EntityType ent = EntityTypes.get(mobName.toLowerCase(Locale.ROOT));
                 if (ent == null) {
-                    throw new NoMatchException("Unknown entity type '" + mobName + "'");
+                    throw new NoMatchException(TranslatableComponent.of("worldedit.error.unknown-entity", TextComponent.of(mobName)));
                 }
                 mobName = ent.getId();
                 if (!worldEdit.getPlatformManager().queryCapability(Capability.USER_COMMANDS).isValidMobType(mobName)) {
-                    throw new NoMatchException("Unknown mob type '" + mobName + "'");
+                    throw new NoMatchException(TranslatableComponent.of("worldedit.error.unknown-mob", TextComponent.of(mobName)));
                 }
-                return validate(context, new MobSpawnerBlock(state, mobName));
+                return new MobSpawnerBlock(state, mobName);
             } else {
                 //noinspection ConstantConditions
-                return validate(context, new MobSpawnerBlock(state, EntityTypes.PIG.getId()));
+                return new MobSpawnerBlock(state, EntityTypes.PIG.getId());
             }
         } else if (blockType == BlockTypes.PLAYER_HEAD || blockType == BlockTypes.PLAYER_WALL_HEAD) {
             // allow setting type/player/rotation
             if (blockAndExtraData.length <= 1) {
-                return validate(context, new SkullBlock(state));
+                return new SkullBlock(state);
             }
 
             String type = blockAndExtraData[1];
 
-            return validate(context, new SkullBlock(state, type.replace(" ", "_"))); // valid MC usernames
+            return new SkullBlock(state, type.replace(" ", "_")); // valid MC usernames
         } else {
-            return validate(context, state.toBaseBlock());
+            return state.toBaseBlock();
         }
     }
 
-    private <T extends BlockStateHolder> T validate(ParserContext context, T holder) {
-        if (context.isRestricted()) {
-            Actor actor = context.requireActor();
-            if (!actor.hasPermission("worldedit.anyblock") && worldEdit.getConfiguration().checkDisallowedBlocks(holder)) {
-                throw new DisallowedUsageException(Caption.toString(Caption.of("fawe.error.block.not.allowed", holder)));
-            }
-            CompoundTag nbt = holder.getNbtData();
-            if (nbt != null) {
-                if (!actor.hasPermission("worldedit.anyblock")) {
-                    throw new DisallowedUsageException("You are not allowed to use nbt'");
-                }
-            }
-        }
-        return holder;
-    }
```

Comment:
I believe removing this also removes the permissions check and config checks here. It could be covered somewhere else but I don't think it is.